packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -217,41 +218,19 @@ class _DropdownMenuRouteLayout<T> extends SingleChildLayoutDelegate {
 
   @override
   Offset getPositionForChild(Size size, Size childSize) {
-    final double buttonTop = buttonRect.top;
-    final double selectedItemOffset = selectedIndex * _kMenuItemHeight + kMaterialListPadding.top;
-    double top = (buttonTop - selectedItemOffset) - (_kMenuItemHeight - buttonRect.height) / 2.0;
-    final double topPreferredLimit = _kMenuItemHeight;
-    if (top < topPreferredLimit)
-      top = math.min(buttonTop, topPreferredLimit);
-    double bottom = top + childSize.height;
-    final double bottomPreferredLimit = size.height - _kMenuItemHeight;
-    if (bottom > bottomPreferredLimit) {
-      bottom = math.max(buttonTop + _kMenuItemHeight, bottomPreferredLimit);
-      top = bottom - childSize.height;
-    }
     assert(() {
       final Rect container = Offset.zero & size;
       if (container.intersect(buttonRect) == buttonRect) {
         // If the button was entirely on-screen, then verify
         // that the menu is also on-screen.
         // If the button was a bit off-screen, then, oh well.
-        assert(top >= 0.0);
-        assert(top + childSize.height <= size.height);
+        assert(menuTop >= 0.0);
+        assert(menuTop + menuHeight <= size.height);
       }
       return true;
     });
 
-    if (route.initialLayout) {
-      route.initialLayout = false;
```

Comment:
This code only ran once, right? I really don't understand why touching the menus broke them.